docs/guides/developer/docs/develop/setup-opencast-for-develop.md

```diff
@@ -89,155 +69,91 @@ install them.
 
-The Homebrew Project adds a package manager to Mac OS. You can install it by:
+#### Assembly profiles
 
-```sh
-$ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
-```
+Besides the default `dist` Maven profile, you can define other profiles to create different builds.
+Activate these profiles using:
 
-You can now install needed packages:
+profile | description
+--------|------------
+`-Pdev`   | Create a single `allinone` distribution and unpack it
+`-Pnone`  | Do not create any distribution
 
-```sh
-$ brew install maven ffmpeg nodejs
-```
+You can only use one of these profiles at a time.
 
-### Git Bash Completion
+#### Build options
 
-In macOS you can not complete or suggest half typed commands with your Tab Key (like you probably know from linux). If you want to use bash completion, you have to install it by
+The options can be used to customize the build process. They can be used in combination with the assembly profiles.
 
-```sh
-$ brew install bash-completion
-```
-
-Find the location of the configuration file
+Option | Description
+-------|------------
+`-DtrimStackTrace=false` | Do not trim stack traces in the logs
+`-DskipTests=true` | Skip tests
+`-T 1.0C` | Use multiple threads for building (Experimental)
 
-```sh
-$ sudo find / -type f -name "git-completion.bash"
-```
+### Build Single Modules
 
-Normally it should be in
+When working on a single Opencast module, it can be extremely helpful to watch the newly built version and include
+it automatically in the Opencast OSGi infrastructure. This can be done through the
+[bundle:watch](https://karaf.apache.org/manual/latest/commands/bundle-watch.html) command in Karaf. The workflow would
+be as follows:
 
-    $ cp /Library/Developer/CommandLineTools/usr/share/git-core/git-completion.bash /usr/local/etc/bash_completion.d/
+* Start Opencast and use `la -u` in the Karaf console to list all installed bundles/modules. Note down the IDs of the
+  bundles you want to watch.
+* Use `bundle:watch IDs` to watch the desired modules, e.g. `bundle:watch 190 199`
+* Make your changes and rebuild the module (e.g. execute `mvn clean install` in the module folder).
+* Watch how Karaf automatically redeploys the changed jars from your local Maven repository. You can verify that
+  everything went smoothly by checking the log with `log:tail`.
 
-Then add following line to the bash_profile in home
+To see this technique in action, you can watch the following short video:
 
-    [ -f /usr/local/etc/bash_completion ] && . /usr/local/etc/bash_completion
+* [Opencast development: Watch and reload modules](https://asciinema.org/a/348132)
 
-Finally apply your changes with
+The updated bundles are only available in the currently running Karaf instance. To create a Opencast version that has
+this changes permanently, you have to run `mvn clean install` in the the assemblies directory again. Your current
+instance will be deleted by the new assembly!
```

Comment:
```suggestion
The updated bundles are only available in the currently running Karaf instance. To create a Opencast version that contains your changes permanently, you have to run `mvn install` in the assemblies directory again. 
```

You do not have to use `clean`, and the currently running instance will not immediately die.